Makefile

```diff
@@ -20,12 +17,37 @@ export KIND_CLUSTER_NAME=kind
 SHELL = /usr/bin/env bash -o pipefail
 .SHELLFLAGS = -ec
 
-build:
-	docker build -f ./operator/Dockerfile -t $(NS)/$(IMAGE_NAME_OPERATOR):$(OPERATOR_TAG) ./operator
-	docker build -f ./nodeagent/Dockerfile -t $(NS)/$(IMAGE_NAME_NODE_AGENT):$(NODE_AGENT_TAG) ./nodeagent
-	docker build -f ./initcontainer/Dockerfile -t $(NS)/$(IMAGE_NAME_INIT_CONTAINER):$(INIT_CONTAINER_TAG) ./initcontainer
-	docker build -f ./samples/netcore/Dockerfile -t $(NS)/$(IMAGE_NAME_NETCORE_SAMPLE):$(NETCORE_SAMPLE_TAG) ./samples/netcore
-	docker build -f ./samples/openarena/Dockerfile -t $(NS)/$(IMAGE_NAME_OPENARENA_SAMPLE):$(OPENARENA_SAMPLE_TAG) ./samples/openarena
+# Support gsed on OSX (installed via brew), falling back to sed. On Linux
+# systems gsed won't be installed, so will use sed as expected.
+SED ?= $(shell which gsed 2>/dev/null || which sed)
+
+GIT_REVISION := $(shell git rev-parse --short HEAD)
+UPTODATE := .uptodate
+# Automated DockerFile building
+# By convention, every directory with a Dockerfile in it will build an image called ghcr.io/playfab/<directory name>
+# An .uptodate file will be created in the directory to indicate that the Dockerfile has been built.
+%/$(UPTODATE): %/Dockerfile
+	@echo
+	docker build --build-arg=revision=$(GIT_REVISION) -t $(NS)thundernetes-$(shell basename $(@D)) -t $(NS)thundernetes-$(shell basename $(@D)):$(IMAGE_TAG) -f $(@D)/Dockerfile .
+	@echo
+	touch $@
+
+# We don't want find to scan inside a bunch of directories, to speed up Dockerfile dectection.
+DONT_FIND := -name vendor -prune -o -name .git -prune -o -name .cache -prune -o -name .pkg -prune -o -name packaging -prune -o -name build-env -prune -o
+
+# Get a list of directories containing Dockerfiles
+DOCKERFILES := $(shell find . $(DONT_FIND) -type f -name 'Dockerfile' -print)
+UPTODATE_FILES := $(patsubst %/Dockerfile,%/$(UPTODATE),$(DOCKERFILES))
+DOCKER_IMAGE_DIRS := $(patsubst %/Dockerfile,%,$(DOCKERFILES))
+IMAGE_NAMES := $(foreach dir,$(DOCKER_IMAGE_DIRS),$(patsubst %,$(NS)thundernetes-%,$(shell basename $(dir))))
+images:
+	$(info $(IMAGE_NAMES))
+	@echo > /dev/null
+
+buildimage: #creates a docker image as a build environment for thundernetes
+	docker build -t thundernetes-src:$(GIT_REVISION) -f build-env/Dockerfile .
+
+build: buildimage $(UPTODATE_FILES)

```

Comment:
The publish.yml pipeline builds the images based on the tag indicated on .versions and then uploads them to GitHub Container Registry. If you run this locally, it should build them but fail to publish them (the secret to authenticate to ghcr.io is in the GitHub Action). However, the publish.yml build step should partially work locally:

`export $(grep -v '^#' .versions | xargs -d '\n') && make build push`

=> exporting to image                                                                                                                                            10.8s
 => => exporting layers                                                                                                                                           10.3s
 => => writing image sha256:58682d09711f3382616816f22edf8a829beb7f07800a191e77cfaa4a918efa0b                                                                       0.0s
 => => naming to docker.io/library/thundernetes-src:02f0fb6                                                                                                        0.1s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
docker push ghcr.io/playfab//thundernetes-operator:
invalid reference format
make: *** [push] Error 1

Reply:
this should be resolved now. I completely missed the push target, though I had it written down on my notepad :(